### PR TITLE
Heatmap: Implementation of color theme

### PR DESCRIPTION
**Pull request by ruchika-narang**
### Description
Implementation of color theme for heatmap

### CODE REVIEW

File: dashboards-observability/common/constants/colors.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    type: 'gradient',
+  },
+  {
+    value: BLUERED_PALETTE.name,
+    title: BLUERED_PALETTE.label,
+    palette: colorPalette(BLUERED_PALETTE.colors, 20),
+    type: 'gradient',
+  },
+  {
+    value: RdBu_PALETTE.name,
```
Comment (abasatwar) [inline dashboards-observability/common/constants/colors.ts]: can be renamed as `redBlue_PALETTE`.

File: dashboards-observability/common/constants/colors.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  colors: ['rgb(0,0,0)', 'rgb(255,255,255)'],
+};
+
+export const BLUERED_PALETTE = {
+  name: 'Bluered',
+  label: 'Blue-Red',
+  colors: ['rgb(0,0,255)', 'rgb(255,0,0)'],
+};
+
+export const RdBu_PALETTE = {
```
Comment (abasatwar) [inline dashboards-observability/common/constants/colors.ts]: can we `RdBu_PALETTE` as `RedBlue_PALETTE`?

File: dashboards-observability/common/constants/colors.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    'rgb(252,78,42)',
+    'rgb(253,141,60)',
+    'rgb(254,178,76)',
+    'rgb(254,217,118)',
+    'rgb(255,237,160)',
+    'rgb(255,255,204)',
+  ],
+};
+
+export const YlGnBu_PALETTE = {
```
Comment (abasatwar) [inline dashboards-observability/common/constants/colors.ts]: same as above.

File: dashboards-observability/common/constants/colors.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    'rgb(29,145,192)',
+    'rgb(65,182,196)',
+    'rgb(127,205,187)',
+    'rgb(199,233,180)',
+    'rgb(237,248,217)',
+    'rgb(255,255,217)',
+  ],
+};
+
+export const DEFAULT_PALETTE = 'default';
```
Comment (abasatwar) [inline dashboards-observability/common/constants/colors.ts:100]: keep this constant at same place in all PRs. This was present in treemap PR. So we can avoid having multiple constants with same name.

File: dashboards-observability/common/constants/colors.ts [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    type: 'gradient',
+  },
+  {
+    value: RdBu_PALETTE.name,
+    title: RdBu_PALETTE.label,
+    palette: colorPalette(RdBu_PALETTE.colors, 20, true),
+    type: 'gradient',
+  },
+  {
+    value: YlOrRd_PALETTE.name,
```
Comment (abasatwar) [inline dashboards-observability/common/constants/colors.ts]: check the names if we can rename.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_heatmap_color_palette_picker.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+}: any) => {
+  const getColorObject = (name: string) => {
+    return {
+      name,
+      color: name,
+    };
+  };
+
+  const heatmapOptions = colorPalettes.filter((color) => color.type !== 'text');
+  console.log(selectedColor, 'selectedColor');
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_heatmap_color_palette_picker.tsx]: please remove console.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_heatmap_color_palette_picker.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+  const onPaletteChange = (value: string) => {
+    onSelectChange(getColorObject(value));
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_heatmap_color_palette_picker.tsx]: `onSelectChange({ name, color: name})` can be used.

File: dashboards-observability/public/components/event_analytics/utils/utils.tsx [hunk 8]
```diff
@@ -285,3 +286,35 @@ export const findAutoInterval = (start: string = '', end: string = '') => {
...
+) => {
+  // default color PLOTLY_COLOR[0]: '#3CA1C7'
+  const defaultColor = [hex, '60', '161', '199'];
+  const rgbElements = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex) || defaultColor;
+  const [, r, g, b] = rgbElements.map((color) => parseInt(color, 16));
+  const rgbaFormat = colorWithOpacity ? `rgba(${r},${g},${b},${opacity})` : `rgb(${r},${g},${b})`;
+  return rgbaFormat;
+};
+
+export const lightenColor = (color: string, percent: number) => {
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/utils/utils.tsx:303]: eslint suppressor is not required?

**Review by abasatwar — CHANGES_REQUESTED** [review]

File: dashboards-observability/common/constants/colors.ts (hunk 4 above)
Comment (ruchika-narang) [reply dashboards-observability/common/constants/colors.ts:100]: It's already on the same line as Treemap.

File: dashboards-observability/public/components/event_analytics/utils/utils.tsx (hunk 8 above)
Comment (ruchika-narang) [reply dashboards-observability/public/components/event_analytics/utils/utils.tsx:303]: The eslint suppressor is present for the whole file, kindly check the first line of this file.

**Review by abasatwar — APPROVED** [review]

**Pull request closed without merging**